server/aesdsocket.c

```diff
@@ -21,48 +21,305 @@
+    	int file_fd = -1;
+    	if (thread_node == NULL)
+    	{
+    	    return NULL;
+    	}
+    	else
+    	{
+    	    node = (socket_node_t *)thread_node;
+    	    file_fd = open(FILENAME, O_CREAT|O_RDWR|O_APPEND, 
+    	                   S_IWUSR | S_IRUSR | S_IWGRP | S_IRGRP | S_IROTH);
+    	    if (file_fd == FAILURE)
+    	    {
+    	        syslog(LOG_ERR, "ERROR: Failed to create/open file");
+    	        status = FAILURE;
+    	        goto exit;
+    	    }
+	
+    	    // Receive data till new line is found
+    	    do
+    	    {
+    	        memset(buffer, 0, BUFF_SIZE);
+    	        recv_bytes = recv(node->accepted_fd, buffer, BUFF_SIZE, 0);
+    	        if (recv_bytes == FAILURE)
+    	        {
+    	            syslog(LOG_ERR, "ERROR: Failed to recieve byte from client");
+    	            status = FAILURE;
+    	            goto exit;
+    	        }
+    	        
+    	        if (pthread_mutex_lock(node->thread_mutex) != SUCCESS)
+    	        {
+    	            syslog(LOG_ERR, "ERROR: Failed to lock mutex");
+    	            status = FAILURE;
+    	            goto exit;
+    	        }
+    	        
+    	        written_bytes = write(file_fd, buffer, recv_bytes);
+    	        if (written_bytes != recv_bytes)
+    	        {
+    	            syslog(LOG_ERR, "ERROR: Failed to write data");
+    	            status = FAILURE;
+    	            pthread_mutex_unlock(node->thread_mutex);
+    	            goto exit;
+    	        }
+    	        if (pthread_mutex_unlock(node->thread_mutex) != SUCCESS)
+    	        {
+    	            syslog(LOG_ERR, "ERROR: Failed to unlock mutex");
+    	            status = FAILURE;
+    	            goto exit;
+    	        }
+    	        // Check if new line
+    	        if ((memchr(buffer, '\n', recv_bytes)) != NULL)
+    	        {
+    	            packet_complete = true;
+    	        }
+    	    } while (!packet_complete);
+
+    	    packet_complete = false;
+
+    	    // Set file pos to begining of file
+    	    off_t offset = lseek(file_fd, 0, SEEK_SET);
+    	    if (offset == FAILURE)
+    	    {
+    	        syslog(LOG_ERR, "ERROR: Failed to SET file offset");
+    	        status = FAILURE;
+    	        goto exit;
+    	    }
+    	    // read till EOF 
+    	    int read_bytes = 0;
+    	    int send_bytes = 0;
+    	    do
+    	    {
+    	        memset(buffer, 0, BUFF_SIZE);
+    	        read_bytes = read(file_fd, buffer, BUFF_SIZE);
+    	        if (read_bytes == -1)
+    	        {
+    	            	syslog(LOG_ERR, "ERROR: Failed to read from file");
+   		    	status = FAILURE;
+              		goto exit;
+            	}
+            	syslog(LOG_INFO, "read succesful is: %d", read_bytes);
+            	syslog(LOG_INFO, "read succesful is: %s", buffer);
+            			
+            	if (read_bytes > 0)
+            	{
+        		// Send file data back to the client
+            	    	send_bytes = send(node->accepted_fd, buffer, read_bytes, 0);
```

Comment:
You would be sending the string with additions characters beyond '\n' here because of your recv & write logic